presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -256,7 +257,7 @@ private static int hashBytes(int initialValue, Slice bytes)
 
     public static Optional<HiveBucketHandle> getHiveBucketHandle(Table table, ConnectorSession session)
     {
-        if (!isBucketExecutionEnabled(session)) {
+        if (!shouldGetBucketInfo(session)) {
```

Comment:
I don't think this is the right place to do this. The `HiveBucketing#getHiveBucketHandle` is used in the `getInsertLayout`. Bucketing cannot be disabled for inserts. If the table is bucketed, we cannot insert into it in non-bucketed way.